src/frontend/src/App.js

```diff
@@ -45,6 +47,20 @@ function App() {
       disable_ocean_mask: formRefs.disableOceanMask.current.checked,
     };
 
+    try {
+      const data = await fetch('/static/taxa_names.json').then((response) => response.json());
+
+      if (!data.includes(formData.taxa_name)) {
+        setBarStatus({loadingStatus: "Invalid Taxa Name", color: "#dc3545"});
```

Comment:
let's also put all possible status message options into a separate constant in dictionary format like:
```js
BAR_STATUS = {
    invalid: {loadingStatus: "Invalid Taxa Name", color: "#dc3545"},
    error: {loadingStatus: "Error checking taxa name", color: "#dc3545"},
     ....
}
```